### ISSUE TITLE
Multi-language dictionary on same keyboard

### ISSUE THREAD

**Issue by BoltCitizen** [dslul/openboard#167]
Thank you for the great keyboard! Beautifully simple and functional!

I am writing in multiple languages. Sometimes within the same message. To get propper spellcheck, I have to always switch the keyboard, which is not practical at all. Maintaining one and same keyboard, but having dictionaries from multiple languages working in parallel would solve this nicely. This is implemented in AnySoftKeyboard and works well.

**Comment by chohner** [comment]
I agree, this would be super useful.
To start, a `Use all dictionaries` toggle inside the `Text correction` setting screen would suffice and maybe not too hard to implement (compared to yet another language picker).

**Comment by vincentmathis** [comment]
This is the only feature that I'm missing from GBoard. Multilingual keyboards, I think GBoard also analyzes the context to guess which language the next word will be for suggestions and spellchecking.

I think it would be enough to have a toggle or option that lets us combine multiple existing dictionaries for suggestions / spell checking. OpenBoard for example has already a german and english dictionary, so maybe a menu where one could select multiple input languages would just combine the contents of the selected dictionaries?

**Comment by jnnkB** [comment]
This may be a duplicate of #7

**Comment by chohner** [comment]
Agreed - they serve the same purpose, although auto-detecting the language is slightly different (and probably harder) than enabling several dictionaries simultaneously

**Comment by Inrumpo** [comment]
This would be a great feature.
But: Is it even possible to use different languages for spell checking (_not:_ word suggestions) _through switching_ keyboards? If yes, #258 might in fact be a bug and not a missing feature?

**Comment by mdxe** [comment]
That would be nice, I always forget to switch keyboard based on the language I use... Unrelated but Firefox 100 spell checking now checks spelling in multiple languages.

**Comment by tdbe** [comment]
BTW devs, in case you're thinking this needs to be solved with language detection, but that is hard and takes time to get right:

Please, please, for now, just add the option to join multiple dictionaries into one. It will solve 80% of the pain. 

And will help you a[dd emojis as one of the dictionaries](https://github.com/openboard-team/openboard/issues/35)!!!!!

**Comment by mrx23dot** [comment]
A simple way:
stick with last language, until seen 4 words in another language (invalid in current one), or user changes languages.

**Comment by Ferk** [comment]
> A simple way:
> stick with last language, until seen 4 words in another language (invalid in current one), or user changes languages.

No, that's actually complicated, it'd have to try and check what language those last 4 words were from... and it wouldn't really work reliably since you won't necessarily have 4 invalid words in the sentence... some languages actually share words (so they might seem valid) and some text messages might even be shorter than 4 words.

Plus, in some conversations I might be mixing words from different languages in the same sentence. In polyglot families that's very common.

It's easier to just have an option to use all dictionaries from the enabled languages at the same time (of course you only enable the languages you know). That would already solve the issue for me, and it would give more predictable results.
